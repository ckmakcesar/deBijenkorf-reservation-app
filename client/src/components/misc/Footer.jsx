import React from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';

import styles from '../../styles/misc/Footer.module.css';

const Footer = ({
  reducedPadding,
  children,
}) => (
  <div className={styles.footerRoot}>
    <hr />

    <div className={clsx(styles.footerWrapper, reducedPadding && styles.footerReducedPadding)}>
      {children}
    </div>
  </div>
);

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
Footer.propTypes = {
  reducedPadding: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Footer.defaultProps = {
  reducedPadding: false,
}

export default Footer;
