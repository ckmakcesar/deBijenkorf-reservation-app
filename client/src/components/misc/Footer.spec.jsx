import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Footer from './Footer';

describe('component:Footer', () => {
  it('Should render Footer with every optional props absent', () => {
    const wrapper = shallow(
      <Footer>
        {/* lint recommends NOT to nest children as prop */}
        <h5>hello world</h5>
      </Footer>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render Footer with every optional props present', () => {
    const wrapper = shallow(
      <Footer reducedPadding={true}>
        <h5>hello world</h5>
      </Footer>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
