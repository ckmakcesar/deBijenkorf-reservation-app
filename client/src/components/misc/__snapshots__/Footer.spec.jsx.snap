// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:Footer Should render Footer with every optional props absent 1`] = `
<div
  className="footerRoot"
>
  <hr />
  <div
    className="footerWrapper"
  >
    <h5>
      hello world
    </h5>
  </div>
</div>
`;

exports[`component:Footer Should render Footer with every optional props present 1`] = `
<div
  className="footerRoot"
>
  <hr />
  <div
    className="footerWrapper footerReducedPadding"
  >
    <h5>
      hello world
    </h5>
  </div>
</div>
`;
